### PR TITLE
Simplify REPL

### PR DESCRIPTION
**Pull request by allanrenucci**
- Remove monadic API when not needed: methods returning `Result[T]` now return `T` if failure is not possible
- Other minor polishing

### CODE REVIEW

File: compiler/test-resources/repl/patdef [hunk 1]
```diff
@@ -22,5 +22,5 @@ val x: Int = 1
 scala> val List(_ @ List(x)) = List(List(2))
 val x: Int = 2
 scala> val B @ List(), C: List[Int] = List()
-val B: List[Int] = Nil
-val C: List[Int] = Nil
+val B: List[Int] = List()
+val C: List[Int] = List()
```
Comment (felixmulder) [inline compiler/test-resources/repl/patdef]: ?

File: library/src/dotty/Show.scala [hunk 2]
```diff
@@ -72,12 +73,12 @@ object Show {
 
   implicit def showList[T](implicit st: Show[T]): Show[List[T]] = new Show[List[T]] {
     def show(xs: List[T]) =
-      if (xs.isEmpty) "Nil"
+      if (xs.isEmpty) "List()"
```
Comment (felixmulder) [inline library/src/dotty/Show.scala]: ?

File: library/src/dotty/Show.scala [hunk 3]
```diff
@@ -72,12 +73,12 @@ object Show {
...
       else "List(" + xs.map(_.show).mkString(", ") + ")"
   }
 
   implicit val showNil: Show[Nil.type] = new Show[Nil.type] {
-    def show(xs: Nil.type) = "Nil"
+    def show(xs: Nil.type) = "List()"
```
Comment (felixmulder) [inline library/src/dotty/Show.scala]: ?

Comment (allanrenucci) [reply library/src/dotty/Show.scala]: It is what scalac does:
```scala
scala> List()
res0: List[Nothing] = List()

scala> List(1, 2)
res1: List[Int] = List(1, 2)
```
I think it is more consistent

Comment (felixmulder) [reply library/src/dotty/Show.scala]: ¯\_(ツ)_/¯

Sounds like a debate for the ages. I'm definitely fine with the change, just wanted to know if there was some other motive beside the superficial one :)

**Pull request merged**